src/sinks/splunk_hec/conn.rs

```diff
@@ -0,0 +1,658 @@
+        };
+
+        let err = build_sink(
+            StubSink::default(),
+            &TowerRequestConfig::default(),
+            &None,
+            &invalid_proxy_config,
+            BatchConfig::default(),
+            Compression::None,
+            Acker::Null,
+            "http://localhost:36448",
+            "token",
+        )
+        .err()
+        .unwrap();
+
+        assert_eq!(
+            err.to_string(),
+            "Failed to build Proxy connector: empty string"
+        )
+    }
+
+    #[derive(Default)]
+    struct StubSink {
+        endpoint: String,
+    }
+
+    #[async_trait::async_trait]
+    impl HttpSink for StubSink {
+        type Input = Vec<u8>;
+        type Output = Vec<u8>;
+
+        fn encode_event(&self, _: Event) -> Option<Self::Input> {
+            Some(String::from("test encoded event").into_bytes())
+        }
+
+        async fn build_request(&self, events: Self::Output) -> crate::Result<Request<Vec<u8>>> {
+            let uri = build_uri(&self.endpoint, "/stub-path")?;
+
+            Request::post(uri).body(events).map_err(Into::into)
+        }
+    }
+}
+
+#[cfg(test)]
+#[cfg(feature = "splunk-integration-tests")]
+mod integration_tests {
+    use super::*;
+    use crate::{assert_downcast_matches, tls::TlsSettings};
+    use integration_test_helpers::get_token;
+    use std::net::SocketAddr;
+    use warp::Filter;
+
+    #[tokio::test]
+    async fn splunk_healthcheck() {
+        let create_healthcheck = |endpoint: String, token: String| {
+            let tls_settings = TlsSettings::from_options(&None).unwrap();
+            let proxy = ProxyConfig::default();
+            let client = HttpClient::new(tls_settings, &proxy).unwrap();
+            super::healthcheck(endpoint, token, client)
+        };
+
+        // OK
+        {
+            let healthcheck =
+                create_healthcheck("http://localhost:8088/".to_string(), get_token().await);
+            healthcheck.await.unwrap();
+        }
+
+        // Server not listening at address
+        {
+            let healthcheck =
+                create_healthcheck("http://localhost:1111".to_string(), get_token().await);
+            healthcheck.await.unwrap_err();
+        }
+
+        // Unhealthy server
+        {
+            let healthcheck =
+                create_healthcheck("http://localhost:5503".to_string(), get_token().await);
+
+            let unhealthy = warp::any()
+                .map(|| warp::reply::with_status("i'm sad", StatusCode::SERVICE_UNAVAILABLE));
+            let server = warp::serve(unhealthy).bind("0.0.0.0:5503".parse::<SocketAddr>().unwrap());
+            tokio::spawn(server);
+
+            assert_downcast_matches!(
+                healthcheck.await.unwrap_err(),
+                HealthcheckError,
+                HealthcheckError::QueuesFull
+            );
+        }
+    }
+}
+
+#[cfg(test)]
+#[cfg(feature = "splunk-integration-tests")]
+pub mod integration_test_helpers {
```

Comment:
This appears to be the same guard as above. Do these helpers need to be in a separate module?

Reply:
I've put these helpers in a separate module, as they are also imported by `sinks::splunk_hec::logs::integration_tests` and `sinks::splunk_hec::metrics::integration_tests`. It seems cleaner to me to have a separate module, rather than having the aforementioned modules import sinks::splunk_hec::conn::integration_tests`